shared/src/commonMain/kotlin/org/hyperskill/app/track_selection/details/view/TrackSelectionDetailsViewStateMapper.kt

```diff
@@ -0,0 +1,87 @@
+package org.hyperskill.app.track_selection.details.view
+
+import org.hyperskill.app.SharedResources
+import org.hyperskill.app.core.view.mapper.NumbersFormatter
+import org.hyperskill.app.core.view.mapper.ResourceProvider
+import org.hyperskill.app.core.view.mapper.SharedDateFormatter
+import org.hyperskill.app.progresses.domain.model.averageRating
+import org.hyperskill.app.track.domain.model.totalTopicsCount
+import org.hyperskill.app.track_selection.details.presentation.TrackSelectionDetailsFeature
+import org.hyperskill.app.track_selection.details.presentation.TrackSelectionDetailsFeature.ContentState
+
+internal class TrackSelectionDetailsViewStateMapper(
+    private val resourceProvider: ResourceProvider,
+    private val dateFormatter: SharedDateFormatter
+) {
+    fun map(state: TrackSelectionDetailsFeature.State): TrackSelectionDetailsFeature.ViewState =
+        when (state.contentState) {
+            ContentState.Idle -> TrackSelectionDetailsFeature.ViewState.Idle
+            ContentState.Loading -> TrackSelectionDetailsFeature.ViewState.Loading
+            is ContentState.Content -> mapContentStateToViewState(state, state.contentState)
+            ContentState.NetworkError -> TrackSelectionDetailsFeature.ViewState.NetworkError
+        }
+
+    private fun mapContentStateToViewState(
+        state: TrackSelectionDetailsFeature.State,
+        contentState: ContentState.Content
+    ): TrackSelectionDetailsFeature.ViewState.Content =
+        with(state.trackWithProgress) {
+            TrackSelectionDetailsFeature.ViewState.Content(
+                title = track.title,
+                description = track.description,
```

Comment:
Let's check for blank `track.description`? Backend `description = models.TextField(default='')`